blog/ugrep-an-interactive-grep-for-the-terminal.md

```diff
@@ -0,0 +1,63 @@
+# ugrep: an interactive grep for the terminal
+
+The ugrep tool is a faster, user-friendly and compatible grep replacement written in C++11. Let's check how to install ugrep and how to use the interactive query mode, fuzzy search, and other options.
+
+## Installation
+
+In Windows, you can just download the full-featured [ugrep.exe](https://github.com/Genivia/ugrep/releases) from the [official repository](https://github.com/Genivia/ugrep), or use a tool like: `choco install ugrep` or `scoop install ugrep`.
+
+In MacOS, you could use `brew install ugrep`.
+
+Or, find your destiny here: <https://github.com/Genivia/ugrep#install>.
+
+## The basics
+
+The basic usage is start the interactive query mode and look for a specific text in recursively in sub-directories using plain-text or regex.
+
+```bash
+ugrep -Q
+```
+
+![ugrep query mode in tldr pages repository](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/b3qsd9og05nypn50vgiv.png)
+
+💡 Tip: you can add the `--ignore-case` argument to make case-insensitive matches.
+
+If you want to see only the files that contains the search term you can use the following arguments:
+
+```bash
+ugrep -Q --ignore-case --files-with-matches
+```
+
+![ugrep query mode listing filenames in tldr pages repository](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/9yllgdg64e2wnv4vegwv.png)
+
+## The fuzzy search
+
+The [fuzzy search](https://en.wikipedia.org/wiki/Approximate_string_matching) is the technique of look for a text that match a pattern approximately, instead of exactly. We can look for a text within the specified [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance).
```

Comment:
```suggestion
The [fuzzy search](https://en.wikipedia.org/wiki/Approximate_string_matching) is a technique that looks for a text that matches a pattern approximately, instead of exactly. We can look for a text within the specified [Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance).
```